artifactory.py

```diff
@@ -1657,17 +1692,23 @@ def find_permission_target(self, name):
             return obj
         return None
 
-    def writeto(self, out, chunk_size=None):
+    def writeto(self, output, chunk_size=1024, progress_func=print_download_progress):
         """
-        Writes artifact to file descriptor in chunks
+        Downloads large file in chunks and and call a progress function.
 
-        :param out: File Descriptor
-        :param chunk_size: Chunk size, default 256
+        :param output: file path of output file
+        :param chunk_size: chunk size in bytes, recommend. eg 1024*1024 is 1MiB
+        :param progress_func: by default print_download_progress. Provide custom function to print output or suppress
+            print by setting to None
+        :return: None
         """
-        if not chunk_size:
-            chunk_size = 256
+        if isinstance(output, str) or isinstance(output, pathlib.Path):
+            context = open(output, "wb")
+        else:
+            context = nullcontext(output)
```

Comment:
Doesn't it work without args? `context = nullcontext()`

Reply:
I went through docs, nullcontext was added in 3.7